### PR TITLE
WFLY-16207 Application caches throw marshalling exceptions when using deployment classes for keys or values

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-16207

### CODE REVIEW

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/infinispan/remote/RemoteInfinispanCacheTestCase.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Deployment(name = DEPLOYMENT_2, managed = false, testable = false)
+    @TargetsContainer(NODE_2)
+    public static Archive<?> deployment2() {
+        return createDeployment();
+    }
+
+    private static Archive<?> createDeployment() {
+        return createDeployment(CreatedCacheBean.class);
```
Comment (rhusar) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/infinispan/remote/RemoteInfinispanCacheTestCase.java:52]: s/CreatedCacheBean/RemoteCacheBean

**Review by rhusar — CHANGES_REQUESTED** [review]

**Comment by rhusar** [comment]
@pferraro The latest failure is:

Caused by: java.security.AccessControlException: WFSM000001: Permission check failed (permission "("javax.management.MBeanPermission" "org.infinispan.client.hotrod.impl.ClientStatistics#-[org.wildfly.clustering.infinispan:cache=org.jboss.as.test.clustering.cluster.infinispan.bean.remote.RemoteCacheBean,name=remote,type=HotRodClient]" "registerMBean")" in code source "(vfs:/content/RemoteCacheBean.war/WEB-INF/classes <no signer certificates>)" of "ModuleClassLoader for Module "deployment.RemoteCacheBean.war" from Service Module Loader")

**Comment by pferraro** [comment]
Adding hold label until we work out how this will play with Infinispan query, etc.

File: clustering/infinispan/embedded/spi/src/main/java/org/wildfly/clustering/infinispan/marshall/DefaultEncoderRegistry.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        }
+    }
+
+    private Transcoder findTranscoder(MediaType fromType, MediaType toType) {
+        return this.transcoderCache.computeIfAbsent(fromType, mt -> new ConcurrentHashMap<>(4)).computeIfAbsent(toType, mt -> this.transcoders.stream().filter(t -> t.supportsConversion(fromType, toType)).findFirst().orElse(null));
+    }
+
+    @Deprecated
+    @Override
+    public org.infinispan.commons.dataconversion.Encoder getEncoder(Class<? extends org.infinispan.commons.dataconversion.Encoder> encoderClass, short encoderId) {
```
Comment (rhusar) [inline clustering/infinispan/embedded/spi/src/main/java/org/wildfly/clustering/infinispan/marshall/DefaultEncoderRegistry.java:105]: Looks like a remnant that can just be an import here and other places (org.infinispan.commons.dataconversion.Encoder) unless it's intentional...

**Review by rhusar — APPROVED** [review]
LGTM now.

File: clustering/infinispan/embedded/spi/src/main/java/org/wildfly/clustering/infinispan/marshall/DefaultEncoderRegistry.java (hunk 2 above)
Comment (pferraro) [reply clustering/infinispan/embedded/spi/src/main/java/org/wildfly/clustering/infinispan/marshall/DefaultEncoderRegistry.java:105]: This was intentional.  The Encoder interface and the methods of EncoderRegistry that use it are deprecated - so the FQCN was used to reduce the likelihood of merge conflicts when these methods are removed.

**Pull request merged**